tools/runtest

```diff
@@ -105,6 +106,39 @@ print_test_info(){
 
 }
 
+# function to run tests. t, tc and total are test name, test count and 
+# total test count respectively. accuracy sets the tolerance. 
+run_test(){
+
+    echo "Running test $tc of $total"
+    cp "$testdir/$t.in" ./
+
+    print_test_info "$t"
+
+    # Run the test case
+    if [ "$buildtype" = 'mpi' ] || [ "$buildtype" = 'cudampi' ] && [ "$ismpirun" = 'yes' ] && [ "$ismp2" = 'no' ]; then
+      $DO_PARALLEL "$qbindir/$qexe" "$t.in" > /dev/null 2>&1
+    else
+      "$qbindir/$qexe" "$t.in" > /dev/null 2>&1
+    fi
+
+    if [ "$?" -ne 0 ]; then
+      echo "Error: $qexe execution failed."
+      echo ""
+      exit 1
+    fi
+
+    # Check the accuracy
+    accuracy='4.0e-3'
```

Comment:
This looks like a dangerously large threshold. Energies should probably agree up to 1.0e-6 Hartree and gradients to 1.0e-4 Hartree/Angstrom.

Reply:
Different thresholds should also be used for dipole moments, Mulliken charges, and other molecular properties.

Reply:
> This looks like a dangerously large threshold. Energies should probably agree up to 1.0e-6 Hartree and gradients to 1.0e-4 Hartree/Angstrom.

@agoetz @vwcruzeiro How come this threshold is ok with Run.tests.amber then?

Reply:
It really depends on the test. Most tests are supposed to test one thing - e.g. the printed energies (in kcal/mol) in the Amber output file along a trajectory. In other cases this is not possible. For instance, to test my DFTB3 implementation in `AmberTools/test/sqm/dftb3` I test differently for single point calculations and geometry optimizations, and grep out stuff that is too sensitive from the geometry optimizations. That is far from ideal but the best I was able to do within the Amber testing framework and my limited time.
```
set SQM = $AMBERHOME/bin/sqm
set SPDIFF = ../../dacdif
set GODIFF = "../../dacdif -a 0.02"
set SPTESTS = (h2o.sp cysdip.sp lignin.sp)
set GOTESTS = (h2o.go cysdip.go)

# single point tests
foreach i ($SPTESTS)
    set input = $i.in
    set output = $i.out
    $SQM -O -i $input -o $output || goto error
    $SPDIFF $output.save $output
end

# geometry optimization tests
# do not check the electronic energy and
# core repulsion energy, they are too sensitive!
# Also, there may be a different number of optimization steps
# on different machines, hence remove xmin output
# Also, do not check the final geometry since small numerical
# differences on different platforms can result in different
# Cartesian coordinates although the internal coordinates are OK
foreach i ($GOTESTS)
    set input = $i.in
    set output = $i.out
    $SQM -O -i $input -o $output || goto error
    grep -v 'Electronic energy' $output > tmp
    grep -v 'Core-core repulsion' tmp > tmp2
    grep -v 'DIPOLE' tmp2 > tmp
    grep -v 'xmin' tmp > tmp2
    sed -e "/Final Structure/,/Calculation Completed/d" tmp2 > $output
    rm tmp tmp2
    $GODIFF $output.save $output
end

exit(0)

error:
echo "  ${0}:  Program error"

```

Reply:
Here is an example of HF based QM/MM MD with Orca (`amber/test/qmmm_EXTERN/QMMM_MD_Orca/Run.aladip.hf_sto-3g`)
```
../../dacdif -t 1 $output.save $output
../../dacdif -t 3 $restrt.save $restrt

```
This was after testing numerical differences on different platforms. I truncate the last digit in the Amber mdout file, leaving energies up to 0.001 kcal/mol ~= 1.e-6 Hartree. And truncate Cartesian coordinates to leave an accuracy of 1.e-4 Angstrom.